onnx_coreml/_operators_nd.py

```diff
@@ -1923,7 +1923,8 @@ def _convert_slice_ir4v9(builder, node, graph, err):
         current_axes = axes[i]
         starts[current_axes] = ip_starts[i]
         ends[current_axes] = ip_ends[i]
-        if ends[current_axes] != INT_MAX or ends[current_axes] < data_shape[current_axes]:
+        if ends[current_axes] < INT_MAX:
```

Comment:
Shouldn't we check with current data shape?

Reply:
yeah that is something which I am not totally clear about. For example: if the data shape (i.e. a particular dimension) is 5 and end is 7, does that mean end is effectively -1 or 2

Reply:
let me add test cases to cover all those cases, so that we know that code is correct

Reply:
just checked the onnx documentation, if end value is larger than the dimension size, its treated as -1. Will update and add a test case.

Reply:
updated it